utility/src/main/java/org/oppia/util/parser/SvgEncoder.kt

```diff
@@ -0,0 +1,43 @@
+package org.oppia.util.parser
+
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import com.bumptech.glide.load.EncodeStrategy
+import com.bumptech.glide.load.Options
+import com.bumptech.glide.load.ResourceEncoder
+import com.bumptech.glide.load.engine.Resource
+import com.caverock.androidsvg.SVG
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.FileOutputStream
+import org.oppia.util.logging.Logger
+
+/** Encodes an SVG internal representation to {@link FileOutputStream}. */
+class SvgEncoder : ResourceEncoder<SVG?> {
+
+  private lateinit var logger: Logger
+  override fun getEncodeStrategy(options: Options): EncodeStrategy {
+    return EncodeStrategy.SOURCE
+  }
+
+  override fun encode(data: Resource<SVG?>, file: File, options: Options): Boolean {
+    return try {
+      val svg: SVG = data.get()
+      val picture = svg.renderToPicture()
+      val bitmap: Bitmap =
+        Bitmap.createBitmap(picture.width, picture.height, Bitmap.Config.ARGB_8888)
+      val canvas = Canvas(bitmap)
+      canvas.drawPicture(picture)
+
+      val os = ByteArrayOutputStream()
+      bitmap.compress(Bitmap.CompressFormat.PNG, /* quality= */100, os)
+      bitmap.recycle()
+      os.writeTo(FileOutputStream(file))
```

Comment:
I don't think that we want this, either. We should be storing the raw SVG file that we downloaded, not a bitmap or binary representation of it. I suspect this actually won't work as-is because the deserializer is expecting decoder is expecting it to be a SVG file, not a PNG.

i see that the SVG class itself doesn't have a way to retrieve its source, so we may need to wrap it in our own Svg class that includes the source & the underlying SVG image, then pass that through the pipeline so that we have access to the original SVG source code here.

Reply:
SvgEncoder is used if Disk caching is necessary in Glide when retrieving data.  If we set DiskCacheStrategy to None we don't need this Encoder and we don't have to manage saving of raw svg file into disk. So I removed this file.

Reply:
Hmm, I actually assumed we would want Glide to cache images but it may not matter if we're already caching images using the AssetRepository. However, that repository is temporary.

Can you add a TODO comment here on #45 to ensure the caching flow is properly hooked up? I want to make sure that if we check in an implementation that explicitly disables caching that we make sure we're properly caching the images ourselves to avoid continuously redownloading images.

Reply:
Sure that seems reasonable I will add a TODO in `GlideImageLoader` file to make sure caching is done proper.

Reply:
Done.